frontend/spec/features/order_prices_spec.rb

```diff
@@ -19,7 +19,7 @@
       click_button 'Add To Cart'
       expect(page).to have_text '$19.99'
       select 'EUR', from: 'currency'
-      wait_for { page.have_text?('€16.00') }
+      wait_for { page.has_text?('€16.00') }
```

Comment:
I've gone and looked further at this - and I believe the issue you're having is that the page before you press 'Add To Cart' already has the text '$19.99' on it - so you have a race condition where the currency can be selected during the Add To Cart action happening.  Instead you need to replace `expect(page).to have_text '$19.99'` with a check for something that isn't on the page until the 'Add To Cart' action has completed.  Something like

    expect(page).to have_css('#link-to-cart', text: '$19.99')

should do it (and then get rid of `wait_for` - it really should never be necessary to use)

Reply:
Thanks for the advise @twalpole . To me it seems like a random error and even after waiting for 60s (which is very long), sometimes those specs didn't pass. That's why I decided to use `wait_for`. But I will try out your solution.

Reply:
@PSzyszkaSpark The reason it was failing previously is because the selection of a new currency was occurring either while the `Add To Cart` action was being processed or before the page was ready to handle it, so no request to change the currency was actually done.  By changing to checking for the text within the specific element it ensures the action has been done and completed (the `link-to-cart` element is loaded via an XHR request after the add to cart action has been completed) so you know the currency selector is ready to be used.

Reply:
Thank you for the explanation. Corrected as you suggested and it works perfectly:)